vault/resource_ldap_secret_backend.go

```diff
@@ -0,0 +1,530 @@
+		data["case_sensitive_names"] = v
+	}
+	if v, ok := d.GetOk("certificate"); ok {
+		data["certificate"] = v
+	}
+	if v, ok := d.GetOk("client_tls_cert"); ok {
+		data["client_tls_cert"] = v
+	}
+	if v, ok := d.GetOk("client_tls_key"); ok {
+		data["client_tls_key"] = v
+	}
+	if v, ok := d.GetOk("groupdn"); ok {
+		data["groupdn"] = v
+	}
+	if v, ok := d.GetOk("insecure_tls"); ok {
+		data["insecure_tls"] = v
+	}
+	if v, ok := d.GetOk("last_rotation_tolerance"); ok {
+		data["last_rotation_tolerance"] = v
+	}
+	if v, ok := d.GetOk("length"); ok {
+		data["length"] = v
+	}
+	if v, ok := d.GetOk("max_ttl"); ok {
+		data["max_ttl"] = v
+	}
+	if v, ok := d.GetOk("password_policy"); ok {
+		data["password_policy"] = v
+	}
+	if v, ok := d.GetOk("request_timeout"); ok {
+		data["request_timeout"] = v
+	}
+	if v, ok := d.GetOk("starttls"); ok {
+		data["starttls"] = v
+	}
+	if v, ok := d.GetOk("tls_max_version"); ok {
+		data["tls_max_version"] = v
+	}
+	if v, ok := d.GetOk("tls_min_version"); ok {
+		data["tls_min_version"] = v
+	}
+	if v, ok := d.GetOk("ttl"); ok {
+		data["ttl"] = v
+	}
+	if v, ok := d.GetOk("upndomain"); ok {
+		data["upndomain"] = v
+	}
+	if v, ok := d.GetOk("url"); ok {
+		data["url"] = v
+	}
+	if v, ok := d.GetOk("use_token_groups"); ok {
+		data["use_token_groups"] = v
+	}
+	if v, ok := d.GetOk("userattr"); ok {
+		data["userattr"] = v
+	}
+	if v, ok := d.GetOk("userdn"); ok {
+		data["userdn"] = v
+	}
+
+	configPath := fmt.Sprintf("%s/config", backend)
+	log.Printf("[DEBUG] Writing %q", configPath)
+	if _, err := client.Logical().Write(configPath, data); err != nil {
+		return diag.FromErr(fmt.Errorf("error writing %q: %s", configPath, err))
+	}
+	log.Printf("[DEBUG] Wrote %q", configPath)
+	return readLDAPConfigResource(ctx, d, meta)
+}
+
+func readLDAPConfigResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	path := d.Id()
+	log.Printf("[DEBUG] Reading %q", path)
+
+	mountResp, err := client.Sys().MountConfig(path)
+	if err != nil && util.Is404(err) {
+		log.Printf("[WARN] %q not found, removing from state", path)
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading %q: %s", path, err))
+	}
+
+	d.Set(consts.FieldBackend, d.Id())
+
+	d.Set(consts.FieldDefaultLeaseTTL, mountResp.DefaultLeaseTTL)
+	d.Set(consts.FieldMaxLeaseTTL, mountResp.MaxLeaseTTL)
+
+	configPath := fmt.Sprintf("%s/config", d.Id())
+	log.Printf("[DEBUG] Reading %q", configPath)
+
+	resp, err := client.Logical().ReadWithContext(ctx, configPath)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading %q: %s", configPath, err))
+	}
+	log.Printf("[DEBUG] Read %q", configPath)
+	if resp == nil {
+		log.Printf("[WARN] %q not found, removing from state", configPath)
+		d.SetId("")
+		return nil
+	}
+
+	if val, ok := resp.Data["binddn"]; ok {
```

Comment:
nit: I was thinking we could simplify all of the below `d.Set` calls to something like what we do in [resource_gcp_auth_backend](https://github.com/hashicorp/terraform-provider-vault/blob/09dc5399b590ef1e58cc00f0ca8548bbf1fcb0b5/vault/resource_gcp_auth_backend.go#L253-L265). But then I found the [resource_gcp_secret_backend sets the state](https://github.com/hashicorp/terraform-provider-vault/blob/09dc5399b590ef1e58cc00f0ca8548bbf1fcb0b5/vault/resource_gcp_secret_backend.go#L162-L166) from the [Sys.ListMounts()](https://github.com/hashicorp/terraform-provider-vault/blob/09dc5399b590ef1e58cc00f0ca8548bbf1fcb0b5/vault/resource_gcp_secret_backend.go#L147) response. If that is better maybe we could do so here? I don't know which is preferred.

Reply:
oh, woops didn't see Vinay already made this recommendation! :)